packages/amplication-server/src/services/segmentAnalytics/segmentAnalytics.service.ts

```diff
@@ -69,12 +75,21 @@ export class SegmentAnalyticsService {
 
   public async track(data: TrackData): Promise<void> {
     if (!this.analytics) return;
+
+    const req = RequestContext.currentContext.req;
+    const analyticsSessionId = req.analyticsSessionId;
+
```

Comment:
It might be easier making this service REQUEST scoped. The performance impact is negligible and we would avoid adding an additional dependency. 
i.e. https://docs.nestjs.com/fundamentals/injection-scopes#request-provider

Reply:
Thanks @overbit 
I tried that initially, but unfortunately, it does not work along with Passport 
https://github.com/nestjs/nest/issues/4581